src/net/sourceforge/kolmafia/AdventureResult.java

```diff
@@ -857,20 +851,18 @@ public String toConditionString() {
    */
   @Override
   public boolean equals(final Object o) {
-    if (!(o instanceof AdventureResult)) {
+    if (!(o instanceof AdventureResult ar)) {
       return false;
     }
 
     if (o instanceof WildcardResult) {
       return o.equals(this);
     }
 
-    AdventureResult ar = (AdventureResult) o;
-
     return this.priority == ar.priority
         && this.id == ar.id
         && (this.name == null || ar.name == null
-            ? this.name == ar.name
+            ? Objects.equals(this.name, ar.name)
```

Comment:
We know that least one name is null. Therefore,  "==" is a perfectly good check that both of them are null.

Why add a method call?